test/smoke/package/src/helpers/testProject.ts

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as path from "path";
+import * as fs from "fs";
+
+export default class TestProject {
+    private _appName: string;
+
+    private _vsCodeConfigPath: string;
+    private _packageJsonPath: string;
+    private _metroConfigPath: string;
+    private _expoSettingsPath: string;
+
+    private _parentPathForWorkspace: string;
+    private _projectEntryPointPath: string;
+    private _testButtonFileForWorkspace: string;
+    private _gradleBuildFilePathForWorkspace: string;
+
+    private _sampleEntryPointPath: string;
+    private _testButtonFileForSample: string;
+    private _gradleBuildFilePathForSample: string;
+
+    constructor(private _workspaceDirectory: string, private _sampleDirectory: string) {
+        this._appName = path.basename(_workspaceDirectory);
+        this._vsCodeConfigPath = path.join(_workspaceDirectory, ".vscode");
+        this._parentPathForWorkspace = path.join(_workspaceDirectory, "..");
+        this._projectEntryPointPath = this.getEntryPoint(_workspaceDirectory);
+        this._sampleEntryPointPath = this.getEntryPoint(_sampleDirectory);
+        this._packageJsonPath = path.join(_workspaceDirectory, "package.json");
+        this._testButtonFileForWorkspace = path.join(this.workspaceDirectory, "AppTestButton.js");
+        this._testButtonFileForSample = path.join(this.sampleDirectory, "AppTestButton.js");
+        this._metroConfigPath = path.join(this.workspaceDirectory, "metro.config.js");
+        this._expoSettingsPath = path.join(this.workspaceDirectory, ".expo", "settings.json");
+        this._gradleBuildFilePathForWorkspace = path.join(
+            this.workspaceDirectory,
+            "android",
+            "app",
+            "build.gradle",
+        );
+        this._gradleBuildFilePathForSample = path.join(this.workspaceDirectory, "build.gradle");
+    }
+
+    private getEntryPoint(diretory: string): string {
+        let entryPoint = path.join(diretory, "App.tsx");
+        if (!fs.existsSync(entryPoint)) {
+            entryPoint = path.join(diretory, "App.js");
+        }
```

Comment:
Maybe it's better to replace this with a ternary operator?
```suggestion
        let entryPoint = fs.existsSync(entryPoint) ? path.join(diretory, "App.tsx") : path.join(diretory, "App.js");
```